kroxylicious/src/main/java/io/kroxylicious/proxy/filter/NetFilter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.filter;
+
+import java.net.SocketAddress;
+
+/**
+ * Abstracts some policy/logic for how an upstream connection for a given client connection
+ * is made.
+ */
+public interface NetFilter {
+
+    /**
+     * Determine the upstream cluster to connect to based on the information
+     * provided by the given {@code context},
+     * by invoking {@link NetFilterContext#initiateConnect(String, int, KrpcFilter[])}.
+     * @param context The context.
+     */
+    void selectServer(NetFilterContext context);
+
+    interface NetFilterContext {
+        /**
+         * @return The address of the client
+         * (possibly not the address of the remote TCP/TLS peer, if any intermediate L4 or
+         * L7 proxy is propagating client connection information).
```

Comment:
Your interpretation is correct. IDK whether using HTTP concepts to illustrate is helpful or not. It's OK here, but in general it could lead people into incorrect reasoning because the protocols are fundamentally different.